scipy/stats/_stats_py.py

```diff
@@ -7913,17 +7915,30 @@ def _count(a, axis=None):
             num = int(num)
     else:
         if axis is None:
-            num = a.size
+            num = xp_size(a)
         else:
             num = a.shape[axis]
     return num
 
 
-def _m_broadcast_to(a, shape):
+def _m_broadcast_to(a, shape, *, xp):
     if np.ma.isMaskedArray(a):
         return np.ma.masked_array(np.broadcast_to(a, shape),
                                   mask=np.broadcast_to(a.mask, shape))
-    return np.broadcast_to(a, shape, subok=True)
+    return xp.broadcast_to(a, shape)
+
+
+def _m_sum(a, *, axis, preserve_mask, xp):
+    if np.ma.isMaskedArray(a):
+        sum = a.sum(axis)
+        return sum if preserve_mask else np.asarray(sum)
+    return xp.sum(a, axis=axis)
+
+
+def _m_mean(a, *, axis, keepdims, xp):
+    if np.ma.isMaskedArray(a):
+        return np.asarray(a.mean(axis=axis, keepdims=keepdims))
+    return xp.mean(a, axis=axis, keepdims=keepdims)
```

Comment:
These and the `_m_` functions above will be removed in two versions, when masked array support is removed.